tests/provider_charm/src/charm.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+# Copyright 2023 Ubuntu
+# See LICENSE file for licensing details.
+#
+# Learn more at: https://juju.is/docs/sdk
+
+"""Charm the service.
+
+Refer to the following tutorial that will help you
+develop a new k8s charm using the Operator Framework:
+
+https://juju.is/docs/sdk/create-a-minimal-kubernetes-charm
+"""
+
+import json
+import logging
+
+import ops
+from charms.ip_router_interface.v0.ip_router_interface import *  # noqa
+
+# Log messages can be retrieved using juju debug-log
+logger = logging.getLogger(__name__)
+
+VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
+
+
+class SimpleIPRouteProviderCharm(ops.CharmBase):
+    """Charm the service."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.RouterProvider = RouterProvides(charm=self)  # noqa
```

Comment:
I think you need the `noqa` because of the name of the variable
```suggestion
        self._router_provider = RouterProvides(charm=self)
```

Reply:
linter is passing without `noqa` and with the same name, so i dont know why that was there haha